### PR TITLE
Fix leaking http connections by `webclient`

### PR DESCRIPTION
**Pull request by rosstimothy**
From the `http.Client` docs:

>CloseIdleConnections closes any connections on its Transport which were previously connected from previous requests but are now sitting idle in a "keep-alive" state. It does not interrupt any connections currently in use.
>If the Client's Transport does not have a CloseIdleConnections method then this method does nothing.
https://pkg.go.dev/net/http#Client.CloseIdleConnections

The `http.Client.Transport` created in `newWebClient` wraps a
chain of `http.RoundTripper` over an underlying `http.Transport`.
Since we cannot guarantee that each `http.RoundTipper` has a
`CloseIdleConnections` method the usage of
`defer clt.CloseIdleConnections()` does not guarantee that the http
connections created during Find/Ping/etc are closed.

To prevent leaking connections implementations of `http.RoundTripper`
have had a `CloseIdleConnections` method added that forwards the
request on to the wrapped interface. The `otelhttp.Transport` does
not implement this method either so care has been taken to wrap it
in a `http.RoundTripper` which will call the root transport in
`enforceCloseIdleConnections`. An upstream issue has been filed
with otelhttp: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/3543
to get the method added to their `Transport` implementation.

This wasn't noticed prior to upgrading to go1.20 because prior to [this](https://github.com/golang/go/commit/4e7e7ae1406c70d9cc0809ec11105a55a60a0b70)
commit the `ReadHeaderTimeout` set on the Proxy web api http.Server
would cause the connection to appear idle and get terminated by the
server.

### CODE REVIEW

**Review by fspmarshall — APPROVED** [review]

File: api/observability/tracing/http/http.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		), rt)
+}
+
+// enforceCloseIdleConnections ensures that the returned [nethttp.RoundTripper]
+// has a CloseIdleConnections method. Since [otelhttp.Transport] does not implement
+// this any [nethttp.Client.CloseIdleConnections] calls result in a noop instead
+// of forwarding the request onto its wrapped [nethttp.RoundTripper].
+//
+// This can be removed once https://github.com/open-telemetry/opentelemetry-go-contrib/issues/3543
+// has been resolved.
```
Comment (marcoandredinis) [inline api/observability/tracing/http/http.go:58]: We should come up with a greppable pattern to find those kind of "DELETE ME when this issue is fixed upstream" :thinking:

File: api/client/webclient/webclient_test.go [hunk 2]
```diff
@@ -442,3 +444,57 @@ func TestSSHProxyHostPort(t *testing.T) {
...
+	srv.Config.ConnState = func(conn net.Conn, state http.ConnState) {
+		stateChange <- state
+	}
+
+	srv.Start()
+	t.Cleanup(srv.Close)
+
+	resp, err := Find(&Config{
+		Context:   context.Background(),
+		ProxyAddr: srv.URL[len("http://"):],
```
Comment (marcoandredinis) [inline api/client/webclient/webclient_test.go:485]:
```suggestion
		ProxyAddr: strings.TrimPrefix(srv.URL, "http://"),
```

File: api/client/webclient/webclient_test.go [hunk 3]
```diff
@@ -442,3 +444,57 @@ func TestSSHProxyHostPort(t *testing.T) {
...
+	require.NoError(t, err)
+	require.Empty(t, cmp.Diff(expectedResponse, resp))
+
+	for _, expected := range expectedStates {
+		select {
+		case state := <-stateChange:
+			require.Equal(t, expected, state, "expected connection state %s got %s", expected.String(), state.String())
+		case <-time.After(10 * time.Second):
+			t.Fatalf("timeout waiting for expected connection state %s", expected.String())
+		}
```
Comment (marcoandredinis) [inline api/client/webclient/webclient_test.go:497]: This test runs really fast (0.007s)
:nail_care: Can we decrease this timeout to something like 1s? Just to increase feedback time if something changes and it starts to fail.

**Review by marcoandredinis — APPROVED** [review]
Just a few nits, no blocker :+1:

**Comment by tigrato** [comment]
@rosstimothy Thank you for the fix.

I think we have some other places where we need to do the same:
- https://github.com/gravitational/teleport/blob/c063352d16b4429b5b7fe7b8242437f882229514/lib/kube/proxy/forwarder.go#L2104
- https://github.com/gravitational/teleport/blob/c063352d16b4429b5b7fe7b8242437f882229514/lib/kube/proxy/kube_creds.go#L92

**Review by marcoandredinis — APPROVED** [review]

**Pull request merged**